fission-core/library/Fission/AWS/Validate.hs

```diff
@@ -0,0 +1,26 @@
+module Fission.AWS.Validate
+  ( validate
+  , HasStatus(..)
+  )
+where
+
+import           RIO
+
+import           Servant
+import           Network.AWS.Route53
+
+import           Fission.Web.Error
+
+
+validate :: HasStatus a => a -> Either ServerError a
+validate res = if status >= 300 then Left (toServerError status) else Right res
+  where status = getStatus res
+
+class HasStatus a where
```

Comment:
A couple recommendations

#  Getting a status code is actually pretty widely applicable
 
* e.g. Client requests also have status codes
* Pull it into its own .Class` module

# Optionally: use `Has`

For example:

```haskell
instance Has StatusCode ChangeResourceRecordSetResponse where
  hasLens = crrsrsResponseStatus
```

# `getStatus` is a bit general

Other libraries make the distinction between a status _message_ and a status _code_

https://www.stackage.org/haddock/lts-16.14/http-types-0.12.3/Network-HTTP-Types-Status.html#t:Status

Maybe name it `StatusCode`?

# Optional: newtype

If you want to get super pedantic, can also wrap in a `StatusCode` newtype wrapper, but that may actually be more hassle than its worth for your use case?

Reply:
nice I like it :+1: 

I ended up doing the `newtype` but did not do `Data.Has` since it seems like we've been kind of moving away from that pattern

Reply:
We are not moving away from `Has` altogether

Reply:
`Has` is more convenient than writing specialized `HasHTTPManager`, `HasFoo`, `HasBar` classes

Reply:
It may be worth chatting about the distinction here when you’re done driving, if you’re interested

Reply:
Yeah definitely interested!